### PR TITLE
#222 Make contributor name and email address editable.

### LINKED ISSUE #222: Allow edit contributors info
Name and Email should be editable.

cursor_and_bitcentive

### PR DESCRIPTION
**Pull request by Alfredo-Delgado**
Resolves #222

v2.1 demo
licecap

v2 demo.
licecap

v1.
licecap

### CODE REVIEW

**Comment by justinbmeyer** [comment]
This is very cool.  But probably overkill.  I'd simply leave the fields as inputs and any change event updates them.  

As part of an example app, it's good to make all of this as stupidly simple as possible.

**Comment by Alfredo-Delgado** [comment]
@justinmeyer I can quickly turn
```
<bit-contributor-field field="name" {(contributor)}="contributor" />
```
into

```
<entity-field {(entity)}="contributor" field="name" />
```

It wasn't my original intent to make a wholly generic component. I don't think this functionality belongs with the rest of the "contributors" logic. As a didactic project, I think it's beneficial to encourage smaller more focused components.

**Comment by justinmeyer** [comment]
@Alfredo-Delgado I think you meant @justinbmeyer

File: public/components/contributors/contributors.stache [hunk 1]
```diff
@@ -10,8 +12,8 @@
   <tbody>
   {{#each contributors as contributor}}
     <tr>
-      <td>{{name}}</td>
-      <td>{{email}}</td>
+      <td><entity-field {(entity)}="contributor" field="name" /></td>
```
Comment (justinbmeyer) [inline public/components/contributors/contributors.stache]: `<model-edit-property {model}="contributor" property="name" />`

File: public/components/entity-field/entity-field.stache [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{{#if isEditing}}
+  <input
+    type="text"
+    class="form-control"
+    {($value)}="fieldValue"
+    ($enter)="updateField()"
+  />
+{{else}}
+  <a href="javascript://" ($click)="makeEditable()" title="Edit" >{{fieldValue}}</a>
```
Comment (justinbmeyer) [inline public/components/entity-field/entity-field.stache]: remove as a link

File: public/components/entity-field/entity-field_test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import QUnit from 'steal-qunit';
```
Comment (justinbmeyer) [inline public/components/entity-field/entity-field_test.js]: make this create its own Model.

**Comment by justinbmeyer** [comment]
Code looks good.  How did you end up styling it?

**Comment by Alfredo-Delgado** [comment]
@justinbmeyer I'd snuck this in to the description:

licecap

No text-decoration and a text cursor.

**Pull request merged**